middleware/locale/locale.go

```diff
@@ -0,0 +1,56 @@
+package locale
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/mholt/caddy/middleware"
+	"github.com/mholt/caddy/middleware/locale/method"
+)
+
+// Locale is a middleware to detect the user's locale.
+type Locale struct {
+	Next     middleware.Handler
+	Locales  []string
+	Methods  []method.Method
+	Settings *method.Settings
+}
+
+// ServeHTTP implements the middleware.Handler interface.
+func (l *Locale) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+	candidates := []string{}
+	for _, method := range l.Methods {
+		candidates = append(candidates, method(r, l.Settings)...)
+	}
+
+	locale := l.firstValid(candidates)
```

Comment:
That does not clarify anything. It's not apparent what this whole section does in its entirety.

I know that it's a habit of young developers, and the German »examine every candidate (for a task)« adds to it, but really avoid the word *candidate* wherevere possible.

Write in idioms like *map*, *sieve*, *get*. And inline as much as possible, only explode (the refactoring is called »extracting«) if something is actually re-used or the original method has grown too big.

I guess you want something like this:
```go
wanted := []BrowserLocale{}
for _, getter := range l.ApplicableGetters { // gets it from HTTP headers, Cookies…
  found := getter(r, l.Conf)
  wanted = append(wanted, found...)
}
sort(wanted) // by precedence

canServe := l.AvailableLocales.Intersect(wanted)
// canServe := l.sieveAvailable(wanted)
// canServe := l.getFirstAvailable(wanted)

if len(canServe) == 0 { // or: if canServe == nil
  canServe = append(canServe, l.DefaultLocale) // or: canServe = l.DefaultLocale
}
r.Header.Set(…, canServe[0]) // or the whole list, or just "canServe" if it's no slice

return l.Next.ServeHTTP(w, r)
```